### ISSUE TITLE
Implicitly generic procs don't perform argument conversion for dependent types

### ISSUE THREAD

**Issue by RedBeard0531** [nim-lang/Nim#7009]
```nim
type Node[T] = object
    val: T

proc genericProc(s: Node; key: s.T) =
    discard # body doesn't matter
proc explicitGenericProc[T](s: Node[T]; key: T) =
    discard # body doesn't matter
proc concreteProc(s: Node[cstring]; key: s.T) =
    discard # body doesn't matter 
    
var strs: Node[cstring]
concreteProc(strs, "string") # string converts to cstring
explicitGenericProc(strs, "string") # still converts
genericProc(strs, "string") # doesn't convert: COMPILE ERROR
```

**Comment by RedBeard0531** [comment]
#7008 was discovered trying to make a minimal repro for this.
#7006 may be related.

**Comment by metagn** [comment]
Works in 2.2, likely same fix as #7006
